_posts/2020-07-21-cube-functionality-to-kartothek.markdown

```diff
@@ -0,0 +1,414 @@
+---
+layout: single
+title:  "Introducing Cube Functionality To Kartothek"
+date:   2020-07-21 06:00:00 +0500
+tags: technology python data-engineering
+header:
+  overlay_image: assets/images/tech_gear_banner.jpg
+  overlay_filter: 0.2
+  show_overlay_excerpt: false
+author: Marco Neumann & Usha Nemani
+author_profile: true
+---
+# Introducing Cube Functionality To Kartothek
+
+In our last blog we introduced [Kartothek](2019-05-28-introducing-kartothek.markdown), a table management python library powered by Dask. 
+Our journey continued by adding a gem to our jewel. We empowered Kartothek with multiple-dataset functionality. 
+Kartothek already provides Dataset features, As a user do I really need a multiple-dataset? Hang On!! 
+Let us spend rest of our time understanding the story of **Cube (kartothek that supports multiple datasets)**. 
+
+## What is a Cube?
+A Cube deals with multiple Kartothek datasets.
+
+Let us start with building a cube for **geodata**. Similar to Kartothek, 
+we need a [simplekv](https://simplekv.readthedocs.io/)-based store backend along with an abstract cube definition.
+
+**df_weather** is a pandas dataframe created from reading a csv file.
+
+```python
+>>> from io import StringIO
+>>> df_weather = pd.read_csv(
+...     filepath_or_buffer=StringIO("""
+... avg_temp     city country        day
+...        6  Hamburg      DE 2018-01-01
+...        5  Hamburg      DE 2018-01-02
+...        8  Dresden      DE 2018-01-01
+...        4  Dresden      DE 2018-01-02
+...        6   London      UK 2018-01-01
+...        8   London      UK 2018-01-02
+...     """.strip()),
```

Comment:
Why use a CSV string to define the dataframe? I would suggest to use a plain dataframe definition using dicts/lists/etc. that's a bit more friendly imo

Reply:
Depends, I found this easier to read than `pd.DataFrame({"x": ..., "y": ...})`, but if you suggest a nice, readable in-code variant, that would of course be better :)